core/handler/convert_fields.go

```diff
@@ -132,8 +132,8 @@ func (f *Functions) Validate(data map[string]interface{}) (bool, error) {
 	return value.ToBoolean()
 }
 
-// Process decodes the specified payload, converts it and test the validity
-func (f *Functions) Process(payload []byte) (map[string]interface{}, bool, error) {
+// ProcessUplink decodes the specified payload, converts it and test the validity
+func (f *UplinkFunctions) ProcessUplink(payload []byte) (map[string]interface{}, bool, error) {
```

Comment:
You're decorating `UplinkFunctions` with this method, so you can just call it `Process`, also for `DownlinkFunctions`